sdk/src/core/wallet_helpers/confirm_pending_transactions.rs

```diff
@@ -0,0 +1,171 @@
+use crate::error::Result;
+use crate::wallet_manager::WalletBorrow;
+use etopay_wallet::{VersionedWalletTransaction, types::WalletTxStatus};
+
+/// Confirm the status of a list of versioned wallet transactions.
+///
+/// This function takes a list of versioned wallet transactions and attempts to confirm their status.
+/// It returns the same list of transactions, with updated entries if the confirmation succeeds.
+///
+/// # Returns
+///
+/// Returns `Ok(Vec<VersionedWalletTransaction>)` containing the original transactions,
+/// with updated status where applicable.
+pub async fn confirm_pending_transactions(
+    wallet: &WalletBorrow<'_>,
+    transactions: &Vec<VersionedWalletTransaction>,
```

Comment:
I suggest we take a mutable slice and mutate the contents in-place here instead of reallocating and cloning everything all the time:

```suggestion
    transactions: &mut [VersionedWalletTransaction],
```

we can then iterate mutably and update the elements one-by-one if needed 💯